sdk/nodejs/x/automation/workspace.ts

```diff
@@ -202,6 +211,14 @@ export interface StackSummary {
     url?: string;
 }
 
+/**
+ * Deployment encapsulates the state of a stack deployment.
+ */
+export interface Deployment {
+    version: number;
+    deployment: object;
```

Comment:
Is there a better type for this? In [Go](https://github.com/pulumi/pulumi/blob/cedab0003bda3b4cbb767d1a8875a620c34ba9da/sdk/go/common/apitype/core.go#L159-L167) we've left this unmarshalled but it's not possible to partially parse a json object in Typescript as far as I'm aware.

Reply:
The type of deployment ends up being one of `DeploymentV1/2/3`: https://github.com/pulumi/pulumi/blob/48f906e0263335ad5fc922f6836b75c5f27da5eb/sdk/go/common/apitype/core.go#L108

That API in go is designed for backwards/forwards compat. SerDe and typing in typescript are far more flexible than in Go, so we may be able to improve the experience here with union types. If you're going to stick with the untyped route that requires the user to do type assertions, `any` may be more idiomatic.

Reply:
Ah perfect, thanks for the pointers. I'll fix this up with a union of the deployment versions.

Reply:
Okay I'm realizing this is a bottomless hole so I'm gonna stick with `any` and leave a note to make it better in the future.